src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1220,7 +1220,11 @@ private void CleanupInheritedProperties()
 					while (propertiesEnumerator.MoveNext())
 					{
 						var dp = propertiesEnumerator.Current;
-						SetValue(dp, DependencyProperty.UnsetValue, DependencyPropertyValuePrecedences.Inheritance);
+						var details = _properties.GetPropertyDetails(dp);
+						if (details.CurrentHighestValuePrecedence == DependencyPropertyValuePrecedences.Inheritance)
```

Comment:
Doing so will likely create memory leaks. Values may be overridden and have a local value (for instance), and the overridden value needs to be cleared regardless.